pinot-query-runtime/src/test/java/org/apache/pinot/query/mailbox/MailboxServiceTest.java

```diff
@@ -610,18 +604,32 @@ public void testRemoteEarlyTerminated()
     sendingMailbox.send(OperatorTestUtil.block(DATA_SCHEMA, new Object[]{0}));
     // receiving-side early terminates after pulling the first block
     TestUtils.waitForCondition(aVoid -> {
-      TransferableBlock block = receivingMailbox.poll();
-      return block != null && block.getNumRows() == 1;
+      MseBlock block = readBlock(receivingMailbox);
+      return block != null && block.isData() && ((MseBlock.Data) block).getNumRows() == 1;
```

Comment:
This change is causing https://github.com/apache/pinot/issues/15580 - we can't assert that the polled block is non-null in `readBlock` here because we're using actual gRPC mailboxes here and the mailbox might not have received the block yet in which case it'd return `null` initially.

Reply:
Nice catch